### ISSUE TITLE
XCursor behaving weird if hyprcursor is enabled

### ISSUE THREAD

**Issue by juspky** [hyprwm/hyprcursor#70]
Today I wanted to play [Oxygen Not Included](https://store.steampowered.com/app/457140/Oxygen_Not_Included/). (Great game btw)
Quickly I realized that the cursor is invisible inside the game window. I suspected hyprcursor so I disabled it and its showing properly now with the xcursor fallback.

Telling from the game files [OSI](https://store.steampowered.com/app/457140/Oxygen_Not_Included/) is some sort of native unity build.
Unfortunately I can't provide further information., but I thought it might be nice to report that there seem to be some compatibility issues with hyprcursor and certain applications.

**Comment by vaxerski** [comment]
I have some serious doubts this is the right repo. That's not how it works. Please gib a hyprland log

**Comment by juspky** [comment]
How does it work then? I'm kinda new to linux and hypr.

All I can tell is that it doesn't work with a hyprcursor theme set but it does work without it

With hyprcursor
https://github.com/user-attachments/assets/e9d08e9e-9f61-47d0-a226-c756618da182
Without hyprcursor
https://github.com/user-attachments/assets/e052d3a5-01a2-4060-93db-8e3986f5bcc6

[hyprlog.md](https://github.com/user-attachments/files/17686957/hyprlog.md)
[hyprland-no-hyprcursor.md](https://github.com/user-attachments/files/17686958/hyprland-no-hyprcursor.md)

**Comment by vaxerski** [comment]
what's the theme? maybe the theme is bork?

Is the game running in native wayland or xwl?

**Comment by tobias-haenel** [comment]
Probably related to https://github.com/hyprwm/Hyprland/issues/7335

**Comment by juspky** [comment]
Initially I had some random theme from discord but I forgot its name. 

To take the video and the log I used https://aur.archlinux.org/packages/hyprcursor-dracula-kde-git 

So I don't think its the themes fault.

The game is running in xwl

**Comment by vaxerski** [comment]
if the game is running in xwayland then I dont think it can use serverside cursors so I doubt this is a hc issue

**Comment by juspky** [comment]
> if the game is running in xwayland then I dont think it can use serverside cursors so I doubt this is a hc issue
you're right about that. after some fiddling around with the hyprcursor theme enabled I realized its not the rendering of the hyprcursor itself but the xcursor fallback which is bugging out. It has nothing to do with OSI, it is just very obvious there.

Check this out.

If I freshly start hyprland without hyprcursor everything works as expected. even after I set the theme.

https://github.com/user-attachments/assets/04f40f07-db70-4292-b5da-a6e86f505811

Now I reboot, cursor is not visible in the header of steam (or OSI).

https://github.com/user-attachments/assets/22017811-a25c-41d4-b8c6-5f1699b749bc
